File: backend/src/main/java/com/example/demo/models/entities/LicitacaoItem.java
package com.example.demo.models.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.util.UUID;

@Table("licitacaoItem")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LicitacaoItem {

    @Id
    private UUID id;
    private UUID licitacaoId;
    private String nome;
    private String descricao;
    private String tratamentoDiferenciado;
    private String aplicabilidadeDecreto7174;
    private String aplicabilidadeMargemDePreferencia;
    private Integer quantidade;
    private String unidadeDeFornecimento;

}

File: scraper/src/app/models/licitacao_item.py
import uuid
from tortoise import fields
from tortoise.models import Model


class LicitacaoItem(Model):
    id = fields.UUIDField(pk=True, default=uuid.uuid4)
    licitacao_id = fields.ForeignKeyField("models.Licitacao", related_name="itens", on_delete=fields.CASCADE)
    nome = fields.CharField(max_length=255, null=True)
    descricao = fields.TextField(null=True)
    tratamento_diferenciado = fields.CharField(max_length=255, null=True)
    aplicabilidade_decreto_7174 = fields.CharField(max_length=255, null=True)
    aplicabilidade_margem_de_preferencia = fields.CharField(max_length=255, null=True)
    quantidade = fields.IntField(null=True)
    unidade_de_fornecimento = fields.CharField(max_length=255, null=True)

    class Meta:
        table = "licitacaoitem"
